db/sql/src/protocol.rs

```diff
@@ -162,13 +164,16 @@ impl HoprDbProtocolOperations for HoprDb {
                             .await?
                     };
 
-                    ChainPacketComponents::into_outgoing(&data, &path, &me, next_ticket, &domain_separator).map_err(
-                        |e| {
-                            crate::errors::DbSqlError::LogicalError(format!(
-                                "failed to construct chain components for a packet: {e}"
-                            ))
-                        },
-                    )
+                    spawn_fifo_blocking(move || {
+                        ChainPacketComponents::into_outgoing(&data, &path, &me, next_ticket, &domain_separator).map_err(
```

Comment:
Why are we doing this in the sql layer?

Reply:
It will need to be done anywhere the async executor needs more room.